src/nft/components/bag/BagFooter.tsx

```diff
@@ -274,31 +294,33 @@ export const BagFooter = ({ totalEthPrice, bagStatus, fetchAssets, eventProperti
   const inputCurrency = useTokenInput((state) => state.inputCurrency)
   const setInputCurrency = useTokenInput((state) => state.setInputCurrency)
   const defaultCurrency = useCurrency('ETH')
+  const inputCurrencyBalance = useTokenBalance(
+    account ?? undefined,
+    !!inputCurrency && inputCurrency.isToken ? inputCurrency : undefined
+  )
 
   const setBagExpanded = useBag((state) => state.setBagExpanded)
   const [tokenSelectorOpen, setTokenSelectorOpen] = useState(false)
 
-  const { balance: balanceInEth } = useWalletBalance()
-  const sufficientBalance = useMemo(() => {
-    if (!connected || chainId !== SupportedChainId.MAINNET) {
-      return undefined
-    }
-    return parseEther(balanceInEth).gte(totalEthPrice)
-  }, [connected, chainId, balanceInEth, totalEthPrice])
-
   const isPending = PENDING_BAG_STATUSES.includes(bagStatus)
   const activeCurrency = inputCurrency ?? defaultCurrency
   const usingPayWithAnyToken = !!inputCurrency && shouldUsePayWithAnyToken
 
   const parsedOutputAmount = useMemo(() => {
     return tryParseCurrencyAmount(formatEther(totalEthPrice.toString()), defaultCurrency ?? undefined)
   }, [defaultCurrency, totalEthPrice])
-  const { state: tradeState, trade, maximumAmountIn } = usePayWithAnyTokenSwap(inputCurrency, parsedOutputAmount)
+  const {
```

Comment:
fast follow: a lot more of this logic should be moved to its own hook/to utils to clean up the Footer component